.eslintrc.js

```diff
@@ -278,7 +278,8 @@ module.exports = {
             ],
           },
         ],
-        'react/prop-types': 'off',
+        'react/prop-types': 'error',
+        'react/no-unused-prop-types': 'error',
```

Comment:
Would it make sense to do the opposite? Disable the rule for `.js` files? Having to disable the rule feels like the exception, not the norm,

Reply:
> I'm not even sure we should disable the rule in the first place to be honest.

Yes, I'm not entirely certain why `packages/mui-base/src/Portal/Portal.js` suddenly fails the rules. compared with e.g. `packages/mui-material/src/Accordion/Accordion.js`, which seems to be be quite similar in setup and is subject to the identical config.

Reply:
Trying that approach in this PR, not sure whether there's any overhead in doing this. Also seems like there are some true positives in packages/mui-material/src/Hidden/HiddenJs.js

Reply:
Just realized another option, we could add

```ts
/* eslint-disable react/no-unused-prop-types */
// ...
/* eslint-enable react/no-unused-prop-types */
```

around the generated prop types

Reply:
As for HiddenJs component - it's an internal one. IMO we shouldn't require proptypes in such cases as developers would see a duplicated error if they use a wrong prop (one from "public" Hidden, the second one from HiddenJs).

Reply:
It will be tricky to selectively turn off `react/prop-types` for internal only components. How would we define the rule in eslint?

I also just reviewed the `HiddenCss` component, and it looks like its proptypes aren't fully matching either.

Reply:
Yes, I suppose we could just disable the rule per file via a comment. Anyway, this can be out of scope of your PR.